Text/TextAnalyticsForHealth/TextAnalyticsForHealth.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using AzureCognitiveSearch.PowerSkills.Common;
+using Azure.AI.TextAnalytics;
+using Azure;
+using System.Globalization;
+
+namespace AzureCognitiveSearch.PowerSkills.Text.TextAnalyticsForHealth
+{
+    public static class TextAnalyticsForHealth
+    {
+        public static readonly string textAnalyticsApiEndpointSetting = "TEXT_ANALYTICS_API_ENDPOINT";
+        public static readonly string defaultTextAnalyticsEndpoint = "https://centralus.api.cognitive.microsoft.com";
+        public static readonly string textAnalyticsApiKeySetting = "TEXT_ANALYTICS_API_KEY";
+        private static readonly int defaultTimeout = 230;
+        private static readonly int maxTimeout = 230;
+        private static readonly int timeoutBuffer = 5;
+        private static readonly int maxCharLength = 5000;
+
+        [FunctionName("TextAnalyticsForHealth")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log,
+            ExecutionContext executionContext)
+        {
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            // Get Endpoint and access key from App Settings
+            string apiKey = Environment.GetEnvironmentVariable(textAnalyticsApiKeySetting, EnvironmentVariableTarget.Process);
+            string apiEndpoint = Environment.GetEnvironmentVariable(textAnalyticsApiEndpointSetting, EnvironmentVariableTarget.Process);
+            if (apiEndpoint == null)
+            {
+                apiEndpoint = defaultTextAnalyticsEndpoint;
+            }
+            if (apiKey == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Healthcare Text Analytics API key is missing. Make sure to set it in the Environment Variables.");
+            }
+            var client = new TextAnalyticsClient(new Uri(apiEndpoint), new AzureKeyCredential(apiKey));
+
+            // Get a custom timeout from the header, if it exists. If not use the default timeout.
+            int timeout;
+            if (!int.TryParse(req.Headers["SkillTimeout"].ToString(), out timeout))
+            {
+                timeout = defaultTimeout;
+            }
+            timeout = Math.Clamp(timeout - timeoutBuffer, 0, maxTimeout);
```

Comment:
>timeout = Math.Clamp(timeout - timeoutBuffer, 0, maxTimeout); [](http://example.com/codeflow?start=12&length=61)

I'm still a little wary about doing the buffer subtraction inside the function. Say the customer requests 300 seconds, then this will come out to 

Math.Clamp(300-5,0,230)=Math.Clamp(295,0,230)=230 

instead of 225 like to actually it actually should be to give the buffer. Also a timeout of 0 isn't very useful tbh. Let's chat at our sync quickly on this.